### ISSUE TITLE
Deep Convolutional Network for Handwritten Chinese Character Recognition

### ISSUE THREAD

**Issue by tengshaofeng** [yuhaozhang/char-rnn-tf#1]
hi,
I am so appreciated with your great job in paper "Deep Convolutional Network for Handwritten Chinese Character Recognition"
but when i write a tensorflow code according to the architecture M7-1 in your paper. It seems not converged. I have try anything i can do such learning rate, Optimizer and so on.

# training process 
training start!
 epoch: 0 db_num: 100/10000 avg_train_loss: 8.3131493187 avg_val_loss:8.23082val_acc:0.0
 epoch: 0 db_num: 200/10000 avg_train_loss: 8.27199922085 avg_val_loss:8.23097val_acc:0.0005
 epoch: 0 db_num: 300/10000 avg_train_loss: 8.25830560048 avg_val_loss:8.2306val_acc:0.0
 epoch: 0 db_num: 400/10000 avg_train_loss: 8.25148080826 avg_val_loss:8.22987val_acc:0.0
 epoch: 0 db_num: 500/10000 avg_train_loss: 8.24735763741 avg_val_loss:8.2292val_acc:0.0005
 epoch: 0 db_num: 600/10000 avg_train_loss: 8.2446211497 avg_val_loss:8.22861val_acc:0.0
 epoch: 0 db_num: 700/10000 avg_train_loss: 8.24270825931 avg_val_loss:8.228val_acc:0.001
 epoch: 0 db_num: 800/10000 avg_train_loss: 8.24126857758 avg_val_loss:8.22774val_acc:0.0
 epoch: 0 db_num: 900/10000 avg_train_loss: 8.24019092454 avg_val_loss:8.22709val_acc:0.0
 epoch: 0 db_num: 1000/10000 avg_train_loss: 8.23933010769 avg_val_loss:8.22652val_acc:0.0
 epoch: 0 db_num: 1100/10000 avg_train_loss: 8.23863611308 avg_val_loss:8.22713val_acc:0.0
 epoch: 0 db_num: 1200/10000 avg_train_loss: 8.23803685904 avg_val_loss:8.22812val_acc:0.0005
 epoch: 0 db_num: 1300/10000 avg_train_loss: 8.23755813965 avg_val_loss:8.22869val_acc:0.0005
 epoch: 0 db_num: 1400/10000 avg_train_loss: 8.23713573796 avg_val_loss:8.22927val_acc:0.0
 epoch: 0 db_num: 1500/10000 avg_train_loss: 8.23680315081 avg_val_loss:8.23023val_acc:0.0
 epoch: 0 db_num: 1600/10000 avg_train_loss: 8.23646358132 avg_val_loss:8.23099val_acc:0.0
 epoch: 0 db_num: 1700/10000 avg_train_loss: 8.23617212576 avg_val_loss:8.23184val_acc:0.0
 epoch: 0 db_num: 1800/10000 avg_train_loss: 8.23592109203 avg_val_loss:8.23268val_acc:0.0

Can you share your code?
thanks so much.

**Comment by yuhaozhang** [comment]
Hi,

This question seems to be irrelevant to this repo.

To answer your question, I was using Caffe to implement the network back then, so there wasn't much of code - all I wrote was the network configuration files and some preprocessing code. Unfortunately I have lost track of even those files due to the long period of time.

I recall that I have encountered the same issue when I trained the networks. This is mainly because when the network gets deeper, the gradients flowing to the early layers tend to be very small. The trick I used was to train a network with fewer layers, and then initialize the earlier layers of the large network with trained parameters from the smaller network. In your case, for example, you can train a 6-layer network with similar network structure, and then initialize the 7-layer network with parameters from the 6-layer network.

On the other hand, there are quite a lot of research recently on adding skip-connections between network layers to improve the training of deep networks, such as the residual network or highway networks. I recommend you to look at some of those papers.

Hope these are helpful,
-Yuhao

**Comment by tengshaofeng** [comment]
thank you for your recommendation.  But when i trained network 6layers, 5layers, it sitll does not work.
less layers can not solve the problem above.
